source/jobs/JobsFeature.h

```diff
@@ -313,6 +317,17 @@ namespace Aws
                      */
                     bool isDuplicateNotification(Iotjobs::JobExecutionData job);
 
+                    /**
+                     * \brief Compares two job documents, ignoring differences in pre-signed URLs.
+                     *
+                     * @param job1 The first job document as a JsonObject
+                     * @param job2 The second job document as a JsonObject
+                     * @return true if the documents are equivalent (ignoring pre-signed URLs), false otherwise
+                     */
+                        
+                    // Add this line to declare the test class as a friend
+                    friend class TestJobsFeaturePrivate;
```

Comment:
Ideally, its better to have no information about the tests in the code we are testing.  

Since `compareJobDocuments` is a utility function that doesn't need access (or be related) to any of the other members of the `JobsFeature` class, it might be a better idea to move this function to a file in the [util folder](https://github.com/awslabs/aws-iot-device-client/tree/main/source/util).

It also helps in testing the `compareJobDocuments` individually, similar to other APIs in the util folder without the need to have a friend class. The mock of the `compareJobDocuments` can be used in the test for `isDuplicateNotification`.

Reply:
Modifications will be made in the revised PR.